src/main/java/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -278,6 +278,16 @@ private static class RefillableState {
     private long lastReportedByteTotal = 0;
 
 
+    /**
+     * @return the given configuration's DataHandler, or null if that DataHandler is a no-op.
+     */
+    private static BufferConfiguration.DataHandler getDataHandler(IonBufferConfiguration configuration) {
+        // Using null instead of a no-op handler enables a quick null check to skip calculating the amount of data
+        // processed, improving performance.
+        BufferConfiguration.DataHandler dataHandler = configuration.getDataHandler();
+        return dataHandler == IonBufferConfiguration.DEFAULT.getDataHandler() ? null : dataHandler;
```

Comment:
It would be nice if we could do something like this. It makes the intention a little more clear.
```suggestion
        return dataHandler == DataHandler.NO_OP ? null : dataHandler;
```